ocaml/xapi/importexport.ml

```diff
@@ -188,7 +188,9 @@ let check_sr_availability ~__context sr =
 let find_host_for_sr ~__context ?(prefer_slaves=false) sr =
   let choose_fn ~host =
     Xapi_vm_helpers.assert_can_see_specified_SRs ~__context ~reqd_srs:[sr] ~host in
-  Xapi_vm_helpers.choose_host ~__context ~choose_fn ~prefer_slaves ()
+  try  
+    Xapi_vm_helpers.choose_host ~__context ~choose_fn ~prefer_slaves ()
+  with _ -> Ref.null
```

Comment:
Please see the first, related comment - it's best not to modify this function and catch the exception it throws instead, because this function is called from multiple places. Therefore it's best not to change the behaviour of this function because we don't know how it will affect the other callers.